src/gather_classes.rs

```diff
@@ -257,80 +254,127 @@ fn update_report(
+                        .get("base")
+                        .ok_or(anyhow!("Different should have base"))?;
+                    let native_list = diff_map
+                        .get("native")
+                        .ok_or(anyhow!("Different should have native"))?;
+
+                    let mut base_calls = HashMap::new();
+                    get_calls_inner(base_list, &mut base_calls)?;
+                    let mut native_calls = HashMap::new();
+                    get_calls_inner(native_list, &mut native_calls)?;
+
+                    merge_calls_with_count(&base_calls, &native_calls, &mut result);
+                } else {
+                    match get_call_key(obj) {
+                        Ok(call_key) => {
+                            result.entry(call_key).and_modify(|c| *c += 1).or_insert(1);
+                        }
+                        // Continue parsing, as there may be other branches in `get_calls_inner` that are valid.
+                        // `obj` may be an object with no key-value pair for "calls".
+                        // For example, if passing a root transaction that failed, `obj` would instead have only the key "revert_reason".
+                        Err(err) => debug!("Failed to extract tuple with error: {err}"),
+                    }
+
+                    if let Some(calls_value) = obj_map.get("calls") {
+                        get_calls_inner(calls_value, result)?;
+                    }
+                }
+                Ok(())
             }
+            _ => Err(anyhow!("unexpected value: `{}`", obj)),
         }
-        _ => Err(anyhow!("unexpected value: `{}`", obj)),
     }
+
+    let mut result = HashMap::new();
+    get_calls_inner(obj, &mut result)?;
+    Ok(result)
 }
 
-/// returns `Result<(EntryPoint, ClassHash), anyhow::Error>`
-fn get_tuple_from_call(call: &Value) -> Result<(EntryPoint, ClassHash), anyhow::Error> {
-    let mut values = Vec::<FieldElement>::new();
-    for key in ["entry_point_selector", "class_hash"] {
-        let next = call
-            .get(key)
-            .context(format!("call: `{call}`\n-------Extracting key {key}"))?;
-        let felt_version = serde_json::from_value(next.clone())
-            .context(format!("convert value `{}` to felt", next))?;
-        values.push(felt_version);
-    }
-    Ok((values[0], values[1]))
+/// Retrieves a [CallKey] from a call object.
+///
+/// If any of the keys are Different, then the call is considered invalid and this function will return an Error.
+fn get_call_key(call: &Value) -> Result<CallKey, anyhow::Error> {
+    let parse_field = |key: &str| -> Result<FieldElement, anyhow::Error> {
+        call.get(key)
+            .and_then(Value::as_str)
+            .ok_or_else(|| anyhow!("Failed to parse call {call} for key: {key}"))
+            .and_then(|s| {
+                if string_is_same(s) {
+                    parse_same_string(s).context(anyhow!("Failed to parse SAME value"))
```

Comment:
If string_is_same parse same should be correct by default, right.

If wouldn't ask, I would directly perform a match:

```rust
match parse_same_string(s) {
   Ok() => "SAME"
   Not Ok() => s
}
```